src/query.c

```diff
@@ -1656,6 +1705,18 @@ int QueryNode_CheckIsValid(QueryNode *n, IndexSpec *spec, RSSearchOptions *opts,
   return res;
 }
 
+// Checks whether query nodes are valid
+// Currently Phrase nodes are checked whether slop/inorder are allowed
+int QAST_CheckIsValid(QueryAST *q, IndexSpec *spec, RSSearchOptions *opts, QueryError *status) {
+  if (!q || !q->root ||
+      !(spec->flags & Index_HasNonEmpty) &&
+      (!isSpecJson(spec) || !(spec->flags & Index_HasUndefinedOrder))
+  ) {
+    return REDISMODULE_OK;
```

Comment:
Might it skip the Tag node check?

Reply:
Notice the `!(spec->flags & Index_HasNonEmpty)` condition